babbage/cube.py

```diff
@@ -131,5 +145,37 @@ def compute_cardinalities(self):
             result = self.members(dimension.ref, page_size=0)
             dimension.spec['cardinality'] = result.get('total_member_count')
 
+    def restrict_joins(self, q, bindings):
+        """
+        Restrict the joins across all tables referenced in the database
+        query to those specified in the model for the relevant dimensions.
+        If a single table is used for the query, no unnecessary joins are
+        performed. If more than one table are referenced, this ensures
+        their returned rows are connected via the fact table.
+        """
+        if len(q.froms) == 1:
+            return q
+        else:
+            for binding in bindings:
+                if binding.table == self.fact_table:
+                    continue
+                concept = self.model[binding.ref]
+                if isinstance(concept, Dimension):
+                    dimension = concept
+                else:
+                    dimension = concept.dimension
```

Comment:
The fact that this assumes the concept is either an attribute or dimension means a cut on a measure might fail. I haven't tested it - it's not a use case for us yet, and I didn't see any unit tests trying that.

Do you see that as a use case or an issue?